File: LeetCode/150EvaluateReversePolishNotation/EvaluateReversePolishNotation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        # Stack -> Time:O(n), Space:O(n)
        s = []
        for i in tokens:
            if i == "+":
                b = s.pop()
                a = s.pop()
                s.append(a+b)
            elif i == "-":
                b = s.pop()
                a = s.pop()
                s.append(a-b)
            elif i == "/":
                b = s.pop()
                a = s.pop()
                if a/b < 0:
                    s.append(math.ceil(a/b))
                else:
                    s.append(math.floor(a/b))
            elif i == "*":
                b = s.pop()
                a = s.pop()
                s.append(a*b)
            else:
                s.append(int(i))
        return s[0]

File: LeetCode/150EvaluateReversePolishNotation/EvaluateReversePolishNotation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        // Stack -> Time:O(n), Space:O(n)
        vector<int> s;
        string plus = "+";
        string minus = "-";
        string division = "/";
        string multiplication = "*";
        int a, b;
        for (int i = 0; i < tokens.size(); i++) {
            if (tokens[i] == plus) {
                a = s[s.size()-2];
                b = s[s.size()-1];
                s.pop_back();
                s.pop_back();
                s.push_back(a + b);
            }
            else if (tokens[i] == minus) {
                a = s[s.size()-2];
                b = s[s.size()-1];
                s.pop_back();
                s.pop_back();
                s.push_back(a - b);
            }
            else if (tokens[i] == multiplication) {
                a = s[s.size()-2];
                b = s[s.size()-1];
                s.pop_back();
                s.pop_back();
                s.push_back(a * b);
            }
            else if (tokens[i] == division) {
                a = s[s.size()-2];
                b = s[s.size()-1];
                s.pop_back();
                s.pop_back();
                s.push_back(a / b);
            }
            else {
                s.push_back(stoi(tokens[i]));
            }
        }
        return s[0];
    }
};
